src/oracle/monitor.js

```diff
@@ -0,0 +1,429 @@
+
+// Create a Registry which registers the metrics
+const register = new Registry()
+
+// Add a default label which is added to all metrics
+register.setDefaultLabels({
+    app: 'agoric-cl-oracle-monitor'
+})
+
+//Create gauge for value
+const oracleSubmission = new Gauge({
+    name: 'oracle_latest_value',
+    help: 'Latest value submitted by oracle',
+    labelNames: ['oracleName', 'oracle', 'feed']
+})
+
+//Create gauge for timestamp
+const oracleObservation = new Gauge({
+    name: 'oracle_last_observation',
+    help: 'Last epoch in which oracle made an observation',
+    labelNames: ['oracleName', 'oracle', 'feed']
+})
+
+//Create gauge for last round
+const oracleLastRound = new Gauge({
+    name: 'oracle_last_round',
+    help: 'Last round in which oracle made an observation',
+    labelNames: ['oracleName', 'oracle', 'feed']
+})
+
+//Create gauge for price deviation
+const oracleDeviation = new Gauge({
+    name: 'oracle_price_deviation',
+    help: 'Latest price deviation by oracle',
+    labelNames: ['oracleName', 'oracle', 'feed']
+})
+
+//Create gauge for balance
+const oracleBalance = new Gauge({
+    name: 'oracle_balance',
+    help: 'Oracle balances',
+    labelNames: ['oracleName', 'oracle', 'brand']
+})
+
+//Create gauge for last price
+const actualPriceGauge = new Gauge({
+    name: 'actual_price',
+    help: 'Actual last price from feed',
+    labelNames: ['feed']
+})
+
+// Register the gaugex
+register.registerMetric(oracleSubmission)
+register.registerMetric(oracleObservation)
+register.registerMetric(oracleLastRound)
+register.registerMetric(oracleBalance)
+register.registerMetric(oracleDeviation)
+register.registerMetric(actualPriceGauge)
+
+const { agoricNames, fromBoard, vstorage } = await makeRpcUtils({ fetch });
+
+//this holds the offer ids
+var feeds = []
+//this holds the amounts in
+var amountsIn = {}
+
+/**
+ * Function to read oracles
+ * @returns oracles, their names and their addresses
+ */
+const readOracleAddresses = () => {
+    var oracles = readJSONFile(ORACLE_FILE)
+    return oracles
+}
+
+/**
+ * Function to get oracles feed invitations
+ */
+export const getOraclesInvitations = async () => {
+    //get the feeds
+    feeds = agoricNames.reverse
+
+    //for each oracle
+    for (let oracle in oracles) {
+
+        const current = await getCurrent(oracle, fromBoard, { vstorage });
+        const invitations = current.offerToUsedInvitation
+
+        //for each invitation
+        for (let inv in invitations) {
+            let boardId = invitations[inv].value[0].instance.boardId
+            let feed = feeds[boardId].split(" price feed")[0]
+
+            if (!("feeds" in oracles[oracle])) {
+                oracles[oracle]["feeds"] = {}
+            }
+            //add feed
+            oracles[oracle]["feeds"][String(inv)] = feed
+        }
+    }
+}
+
+//var oracleLabels = readOracles();
+var oracles = readOracleAddresses();
+await getOraclesInvitations();
```

Comment:
I suggest we don't add logic in between function and variable definitions.

Reply:
This also applies to this 
![image](https://user-images.githubusercontent.com/23262259/220381603-4d9e52d2-3bf6-497e-b627-f720d7087df5.png)

Reply:
Fixed in https://github.com/jacquesvcritien/agoric-cl-middleware/commit/bafa5748c9af7238d24b465797fdb075b783a8b0